tests/components/evohome/test_setup.py

```diff
@@ -0,0 +1,116 @@
+"""The tests for evohome."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import pytest
+
+from homeassistant.components.climate import ATTR_CURRENT_TEMPERATURE, ATTR_PRESET_MODE
+from homeassistant.components.evohome import DOMAIN
+from homeassistant.components.water_heater import ATTR_AWAY_MODE, ATTR_OPERATION_MODE
+from homeassistant.const import ATTR_SUPPORTED_FEATURES, ATTR_TEMPERATURE
+from homeassistant.core import HomeAssistant, State
+from homeassistant.helpers import entity_registry as er
+
+from .conftest import expected_results_fixture, setup_evohome
+from .const import TEST_INSTALLS
+
+_LOGGER = logging.getLogger(__name__)
+
+CORE_STATE_ATTRS = (
+    ATTR_AWAY_MODE,
+    ATTR_CURRENT_TEMPERATURE,  # climate and water_heater
+    ATTR_OPERATION_MODE,
+    ATTR_PRESET_MODE,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+)
+
+
+class ExpectedResults:
```

Comment:
Please remove this class. Using assertion helpers rarely makes the tests easier to read. Use snapshots for testing the entity state. The services can simply be asserted in each test. We don't need a helper for that. I suggest using pytest parameters.